### DISCUSSION TITLE
time widget

### DISCUSSION THREAD

**Discussion by Vader77Darth** [zepp-health/zeppos-forum#136] [Q&A]
I am trying to make a time widget for mi band 7.
I want the time to be 12 hour time. dont want it to create zeroes infront of the hour. And I want the whole time widget to be on one line and centered. Any ideas how?

**Comment by MyoungXUE** [comment]
Hi, I can provide some ideas.

Use the TIME API to get the time. https://docs.zepp.com/docs/1.0/reference/device-app-api/hmSensor/sensorId/TIME/

Use the TEXT widget for display and set the `align_h` to `hmUI.align.CENTER_H`. https://docs.zepp.com/docs/1.0/reference/device-app-api/hmUI/widget/TEXT/

**Reply by Vader77Darth** [reply]
Any idea how I could still be able to use custom fonts like you can with the time widgets and date widgets for example?

**Reply by MyoungXUE** [reply]
> Any idea how I could still be able to use custom fonts like you can with the time widgets and date widgets for example?

Unfortunately, mi band 7 does not support custom fonts.

**Reply by Vader77Darth** [reply]
I guess if I want custom font, I will have to use the time widget and lose the ability to centre it with only 3 digits for time.

**Reply by MyoungXUE** [reply]
@Vader77Darth There is another idea, if only the numbers need to use fonts, you can prepare ten images of numbers 0-9 and use IMG widget to display them.

**Reply by Vader77Darth** [reply]
So, how would I achieve this exactly? I would be able to get the time as IMG widgets but how would I centre the whole group of IMG widgets?

**Reply by MyoungXUE** [reply]
The width and height of each image is usually a certain size, so it is easy to center the image by calculating the width and height of the screen and setting the corresponding coordinates.

For the implementation of this part, you can refer to the watchface, such as the time at the top of the basketball watchface
https://docs.zepp.com/docs/samples/watchface/basketball/

https://github.com/zepp-health/zeppos-samples/tree/main/watchface/1.0/basketball

**Reply by Vader77Darth** [reply]
But this will all change completely when the time goes from 9:59 to 10:00 as 10:00 has another digit. The centre of each image would have to move.

**Reply by MyoungXUE** [reply]
This is the case, so the time is usually designed to fill in the front with zero.

**Reply by Vader77Darth** [reply]
I don't want  zero at the front. If I wanted that, I could have easily used the time widget. The difficulty is centering the widget when you don't have a zero at the front.
